src/lib/features/scheduler/scheduler-service.ts

```diff
@@ -28,13 +29,23 @@ export class SchedulerService {
     private executingSchedulers: Set<string> = new Set();
 
     constructor(
-        getLogger: LogProvider,
+        {
+            getLogger,
+            gracefulShutdown,
+            eventBus,
+        }: Pick<IUnleashConfig, 'getLogger' | 'gracefulShutdown' | 'eventBus'>,
         maintenanceStatus: IMaintenanceStatus,
-        eventBus: EventEmitter,
     ) {
         this.logger = getLogger('/services/scheduler-service.ts');
         this.maintenanceStatus = maintenanceStatus;
         this.eventBus = eventBus;
+        gracefulShutdown.registerGracefulShutdownHook(
+            'scheduler-service',
+            async () => {
+                this.logger.info('Shutting down');
```

Comment:
will remove this line, it was here to see that we did log with the correct logger during implementation phase.